test/validation/api/traffic_mngr/traffic_mngr.c

```diff
@@ -386,35 +385,31 @@ static odp_bool_t approx_eq64(uint64_t val, uint64_t correct)
 static int test_overall_capabilities(void)
 {
 	odp_tm_level_capabilities_t *per_level;
-	odp_tm_capabilities_t        capabilities_array[MAX_CAPABILITIES];
+	odp_tm_capabilities_t        capabilities_array[2];
 	odp_tm_capabilities_t       *cap_ptr;
+	odp_tm_egress_t              egress;
 	uint32_t                     num_records, idx, num_levels, level;
 	int                          rc;
 
-	rc = odp_tm_capabilities(capabilities_array, MAX_CAPABILITIES);
-	if (rc < 0) {
-		CU_ASSERT(rc < 0);
-		return -1;
-	}
+	odp_tm_egress_init(&egress);
+	egress.egress_kind = ODP_TM_EGRESS_PKT_IO;
+	egress.pktio = xmt_pktio;
+
+	rc = odp_tm_egress_capabilities(&capabilities_array[0], &egress);
+	CU_ASSERT_FATAL(rc == 0);
+	num_records = 1;
 
-	/* Now test the return code (which did not indicate a failure code)
-	 * to make sure that there is at least ONE capabilities record
-	 * returned */
-	if (rc == 0) {
-		CU_ASSERT(rc != 0);
+	/* Get capabilities for egress kind function. */
+	odp_tm_egress_init(&egress);
+	egress.egress_kind = ODP_TM_EGRESS_FN;
+	rc = odp_tm_egress_capabilities(&capabilities_array[1], &egress);
+	CU_ASSERT(rc == 0);
+	if (rc < 0)
```

Comment:
This could also be changed the same way as the similar condition little earlier, i.e. we want to stop the test if the capabilities call returns unexpected return code, even if it is positive.

Otherwise this PR looks ok to. With the above change, Reviewed-by: Janne Peltonen <janne.peltonen@nokia.com>